signer/src/bitcoin/mod.rs

```diff
@@ -147,4 +147,30 @@ pub trait BitcoinInteract: Sync + Send {
     fn get_network_info(
         &self,
     ) -> impl Future<Output = Result<bitcoincore_rpc_json::GetNetworkInfoResult, Error>> + Send;
+
+    /// Gets the best (canonical, chain tip from chain with most work) block hash from the Bitcoin node.
+    fn get_best_block_hash(&self) -> impl Future<Output = Result<BlockHash, Error>> + Send;
+}
+
+/// A trait for providing a stream of block hashes to be used by the block observer.
+///
+/// Implementors of this trait are responsible for sourcing block hash notifications,
+/// such as new chain tips from a Bitcoin Core node, and making them available as an
+/// asynchronous stream. This abstraction allows different components, such as
+/// block observers or test utilities, to consume block hash events without being
+/// coupled to the specific mechanism of how those events are obtained.
+pub trait BitcoinBlockHashStreamProvider: Send + Sync {
+    /// The error type that this provider can return.
+    type Error: std::error::Error;
```

Comment:
I suppose it doesn't hurt, but why give the choice? Seems simpler to always return a specific error. In this case, we should just return the crate error here.

Reply:
Same as above, it can't be the crate `Error` with broadcast streams. The primary reason for the associated type is to allow test implementations use the error most convenient/appropriate for their context.

Reply:
Right, we need clone for broadcast streams.